### ISSUE TITLE
Lego-Mathworks interface

### ISSUE THREAD

**Issue by clplaneguy** [jakubk28/LegoMindstormsEV3#1]
How is your work going with the GyroBoy using Mathworks software?  That is what I am trying to do.  But, your code goes beyond Mathworks and is not working for me.  I hope you will be doing more.

**Comment by thh22** [comment]
This is still in development and I do hope to update the repository in due course. Are you able to describe your issue in more detail? Have you successfully opened the Simulink file? It is worth trying some of the Simulink examples from their support package for Lego Mindstorms to test the Wifi connection - particularly this one: https://uk.mathworks.com/help/supportpkg/legomindstormsev3/examples/communicating-with-lego-mindstorms-ev3-hardware.html?prodcode=SL

**Comment by clplaneguy** [comment]
I am glad to hear from you.

I have put much work into Lego, MakeBlock and MathWorks and gotten absolutely nowhere towards my final goal.  Lego is introductory.  MakeBlock has more powerful hardware.  Now, I have moved on to MathWorks.  MathWorks provides no support for MakeBlock but does for Lego.  So, I will fall back to Lego so that I can work with MathWorks.  

**Post 6/14/2019  StackExchange(SuperUser)**
MathWorks (MATLAB and Simulink) is(are) written for Lego firmware 1.03! (with no TelNet ID/passkey) Versions 1.07 and 1.08 also do not require the ID/passkey. Versions 1.09 and 1.10 do require an ID/passkey. MathWorks software does not provide an ID/passkey and thus only works with V1.03, V1.07 or V1.08! MathWorks does not work with the currently available V1.09 or V1.10 and Lego does not provide the old firmware (1.03, 1.07 or 1.08). It should still exist somewhere. In order for me to get it, it must be provided by the general public.

**Can someone please post Lego firmware V1.03, V1.07 or V1.08?**

PS.  My 30 day trial ends in two days and nether MathWorks nor I are able to create an EV3 object by either USB or Wi-Fi.  As a beginner, I have no experience with Telnet.  But, USB should still work if set up properly.  We have not even been able to get a USB connection to work.

**Comment by thh22** [comment]
Have you tried following the setup guide I've put on this github page? I am using the developer version of the Lego firmware - v1.09D - and have edited the code in the Simulink support package to send a password for telnet connection. This works ok on Windows 10 with Matlab 2018a and the Edimax USB adaptor - although I do have intermittent Wifi connection issues. I've notified the Mathworks about the need to send a password so hopefully they will implement something similar to this fix in the future.

I suggest to try following the setup_guide.pdf file and see how you get on.

**Comment by clplaneguy** [comment]
The old firmware does not work with the new Lego software.  It  was
MathWorks that gave me your link.  I have edited the legoev3.m file
according to your instructions.  And an ev3shell.m file.  Lego is up to
firmware V1.10 and I have upgraded to this.  I have not gotten the
ID/passkey to work with V1.09D or V1.10!  I have not gotten the ID/passkey
to work with USB.  This issue is now on record with both Lego and
MathWorks.  If the expert programmers at MathWorks cannot resolve this
issue in 30 days, I thank perhaps they do not consider Lego to be important.
